src/SSVOpenHexagon/Core/HGGraphics.cpp

```diff
@@ -228,11 +266,19 @@ void HexagonGame::draw()
             }
 
             // Draw wall layers
-            for(std::size_t k = j * numWallQuads; k < (j + 1) * numWallQuads;
-                ++k)
+            for(std::size_t k = jAdj * numWallQuads;
+                k < (jAdj + 1) * numWallQuads; ++k)
             {
-                wallQuads3D[k].position += newPos;
-                wallQuads3D[k].color = overrideColor;
+                if(renderingAbove)
+                {
+                    wallQuads3DTop[k].position += newPos;
+                    wallQuads3DTop[k].color = overrideColor;
+                }
+                else
+                {
+                    wallQuads3D[k].position += newPos;
+                    wallQuads3D[k].color = overrideColor;
+                }
```

Comment:
Pull the `if` statement outside of the loop, so that we have `if (renderAbove) { for{...} } else { for{...} }`.